### DISCUSSION TITLE
Tune mode adjust seed

### DISCUSSION THREAD

**Discussion by adrianferal-is** [ultralytics/ultralytics#9859] [Q&A] [answered]
Hello, is it possible to adjust the seed from where the tune option starts the search? I ve worked with yolov7 in the past and that was an option but i have not found the way to do that in yolov8

thanks for your help !

**Comment by glenn-jocher** [comment]
Hello! Thanks for reaching out with your question. 😊 Yes, it's indeed possible to set the random seed in YOLOv8, much like you might have done with YOLOv7. While the `tune` mode's options might initially seem different, you can actually set the seed globally for reproducibility across all operations, including tuning, training, and validation. 

To set the seed in your command, you can simply add the `--seed` argument followed by your desired seed value. Here's a quick example:

```bash
yolo detect train --data coco128.yaml --model yolov8n.yaml --epochs 100 --seed 42
```

This ensures that all the random processes start from the same point, allowing for a consistent start for the tuning process. Let me know if this helps or if there's anything else I can assist you with!

**Reply by adrianferal-is** [reply]
Thanks for the quick answer, i am using the yolo api so my code looks like this: 
    model = YOLO('D:/Datasets/yolov8x-cls.pt')
    #dataset_path ='D:/Datasets/' 
    dataset_path ='D:/Datasets/d1/'
    results = model.tune(data=dataset_path, epochs=150, imgsz=256, batch=32, workers=8, iterations=50, optimizer='AdamW')
  
i wanted to set the start of the evolve to the parameters of the best training i had so it doesnt start from zero each time we reset the tune model, is this possible?

Another question where can i search for more info regarding the --seed parameter?. what is that 42 number, a simple number that sets the experiments for repeatibility?

thanks a lot for the help you are doing a great job with the repository

**Reply by glenn-jocher** [reply, marked as answer]
Hello and thanks for your kind words! 😊

To resume tuning from the best known parameters, you can use the `resume` argument in your code. If you've saved your best model as `best.pt`, for example, you can add `resume='path/to/best.pt'` to your `tune()` method:

```python
results = model.tune(data=dataset_path, epochs=150, imgsz=256, batch=32, 
                     workers=8, iterations=50, optimizer='AdamW', 
                     resume='D:/Datasets/best.pt')
```
This will start the tuning process with parameters from your best-performed training session. 

Regarding the `--seed` parameter, indeed, the number 42 (or any other number you choose) is used for ensuring reproducibility across runs. It seeds the random number generators allowing for the experiments to be repeatable. So, in effect, using the same seed with the same inputs and configurations will give you identical outputs each time.

You can find more detailed information about the seed and other parameters in the [Ultralytics YOLOv8 Documentation](https://docs.ultralytics.com) or directly within the repository's README.

Hope this helps, and feel free to reach out if you have more questions! 🚀

**Reply by adrianferal-is** [reply]
I ve changed that so that my training is resume with the best weights however it pops an error:FileNotFoundError: [WinError 3] the system can not find the path specified: 'runs\\classify\\train\\weights'

**Reply by glenn-jocher** [reply]
Hello! 😊 It looks like the issue may be related to the path specified for resuming training with the best weights. Could be that the directory doesn’t exist or there’s a slight mismatch in the path. Please make sure that the `resume` path you're providing points exactly to the `.pt` file of your best model weights. For instance:

```python
resume='D:/Datasets/runs/classify/train/weights/best.pt'
```

Ensure that the path up to `best.pt` is correct and accessible. Let us know if this solves the problem or if you need further assistance!

**Reply by adrianferal-is** [reply]
im still getting the same error, could it be that my directory isnt like the one specified in the example, i have some train results i want to start from. Here is a screenshot of the directory i want to start from, thing is when i pass the .pt file in the constructor of YOLO(<".pt file path">) it does start the evolve.
![image](https://github.com/ultralytics/ultralytics/assets/118820735/c02e1b4a-96c1-499c-b272-0bca364b496c)

**Reply by glenn-jocher** [reply]
Hey there! Thanks for reaching out with your issue. 😊 It seems like there might be a bit of confusion about where and how to specify the `.pt` file path for resuming your training to evolve the model further.

From what you've described, it sounds like you're trying to load the weights directly into the YOLO constructor which is indeed the correct approach for starting a new model with pre-trained weights. If you're looking to *evolve* from those weights though, the key is making sure the `resume` argument points directly to your `.pt` file when calling the `tune()` method, rather like this:

```python
model = YOLO('D:/Datasets/yolov8x-cls.pt')  # Loads model with initial weights for tuning
results = model.tune(data='D:/Datasets/d1/', 
                     epochs=150, 
                     imgsz=256, 
                     batch=32, 
                     workers=8, 
                     iterations=50, 
                     optimizer='AdamW', 
                     resume='D:/Correct/Path/To/Your/weights/best.pt')  # specify correct path
```

Double-check the path to your `best.pt` file, ensuring it's accessible and correctly formatted. The path error suggests the file isn’t being found where expected.

Hope this helps clear things up! Let us know if there’s anything else we can help with.
